packages/webapp/src/components/Modals/WaterUseCalculatorModal/index.jsx

```diff
@@ -1,32 +1,315 @@
+  const APPLICATION_DEPTH_UNIT = 'application_depth_unit';
+  const DEFAULT_LOCATION_APPLICATION_DEPTH = 'default_location_application_depth';
+  const PERCENTAGE_LOCATION_IRRIGATED = 'percentage_location_irrigated';
+  const PERCENTAGE_LOCATION_IRRIGATED_UNIT = 'percentage_location_irrigated_unit';
+  const LOCATION_SIZE = 'location_size';
+  const LOCATION_SIZE_UNIT = 'location_size_unit';
+  const IRRIGATED_AREA = 'irrigated_area';
+  const IRRIGATED_AREA_UNIT = 'irrigated_area_unit';
+
+  useEffect(() => {
+    if (irrigated_area) {
+      setTotalWaterUsage(() => irrigated_area * application_depth);
+    }
+  }, [irrigated_area, application_depth]);
+
+  useEffect(() => {
+    if (locationSize && percentage_location_irrigated) {
+      const irrigatedArea =
+        locationSize * (percentage_location_irrigated ? percentage_location_irrigated / 100 : 1);
+      setValue(IRRIGATED_AREA, irrigatedArea);
+    }
+  }, [locationSize, percentage_location_irrigated]);
+
+  return (
+    <>
+      <Unit
+        register={register}
+        displayUnitName={APPLICATION_DEPTH_UNIT}
+        label={t('ADD_TASK.IRRIGATION_VIEW.ESTIMATED_APPLICATION_DEPTH')}
+        hookFormSetValue={setValue}
+        hookFormGetValue={getValues}
+        hookFromWatch={watch}
+        name={APPLICATION_DEPTH}
+        unitType={irrigation_depth}
+        max={999999.99}
+        system={system}
+        control={control}
+        selectDisabled={true}
+      />
+
+      <Checkbox
+        label={t('ADD_TASK.IRRIGATION_VIEW.DEFAULT_APPLICATION_DEPTH')}
+        sm
+        style={{ marginTop: '10px', marginBottom: '30px' }}
+        hookFormRegister={register(DEFAULT_LOCATION_APPLICATION_DEPTH)}
+      />
+      <Unit
+        register={modalState.register}
+        displayUnitName={PERCENTAGE_LOCATION_IRRIGATED_UNIT}
+        label={t('ADD_TASK.IRRIGATION_VIEW.PERCENTAGE_LOCATION_TO_BE_IRRIGATED')}
+        hookFormSetValue={modalState.setValue}
+        hookFormGetValue={modalState.getValues}
+        hookFromWatch={modalState.watch}
+        name={PERCENTAGE_LOCATION_IRRIGATED}
+        unitType={percentage_location}
+        max={100}
+        system={system}
+        control={modalState.control}
+      />
+
+      <div
+        style={{
+          flexDirection: 'row',
+          display: 'inline-flex',
+          paddingBottom: '40px',
+          width: '100%',
+          gap: '16px',
+          marginTop: '20px',
+        }}
+      >
+        <Unit
+          register={register}
+          classes={{ container: { flexGrow: 1 } }}
+          label={t('ADD_TASK.IRRIGATION_VIEW.LOCATION_SIZE')}
+          name={LOCATION_SIZE}
+          displayUnitName={LOCATION_SIZE_UNIT}
+          unitType={location_area}
+          system={system}
+          hookFormSetValue={setValue}
+          hookFormGetValue={getValues}
+          hookFromWatch={watch}
+          control={control}
+          required
+          value={locationSize}
+          disabled={true}
```

Comment:
just write disabled no need to write disabled={true}. as you can see 'required' is also written the same way